app/controllers/posts_controller.rb

```diff
@@ -1,6 +1,18 @@
 class PostsController < ApplicationController
+  PAGE_SIZE = 5
+
   def index
-    @posts = Post.all
+    if params[:sort] == 'top'
+      @posts = Post.all
+      descending = -1
+      @posts = @posts.sort_by do |post|
+        post.post_points * descending
+      end
+    else
+      @page = (params[:page] || 0).to_i
+      @posts = Post.offset(PAGE_SIZE * @page).limit(PAGE_SIZE)
```

Comment:
Does this only paginate if there isn't any sorting?

Reply:
That's how it is right now. I wasn't sure how to paginate after already having all the posts sorted. I'm looking into it right now

Reply:
Also I set up both mainly to practice both things, but am looking at combining

Reply:
Done 🔥